internal/mode/static/nginx/runtime/manager.go

```diff
@@ -17,18 +17,50 @@ import (
 )
 
 const (
-	pidFile            = "/var/run/nginx/nginx.pid"
-	pidFileTimeout     = 10000 * time.Millisecond
-	nginxReloadTimeout = 60000 * time.Millisecond
+	// PidFile specifies the location of the PID file for the Nginx process
+	PidFile = "/var/run/nginx/nginx.pid"
+	// pidFileTimeout defines the timeout duration for accessing the PID file
+	pidFileTimeout = 10000 * time.Millisecond
+	/// NginxReloadTimeout sets the timeout duration for reloading the Nginx configuration
+	NginxReloadTimeout = 60000 * time.Millisecond
 )
 
 type (
-	readFileFunc  func(string) ([]byte, error)
-	checkFileFunc func(string) (fs.FileInfo, error)
+	ReadFileFunc  func(string) ([]byte, error)
+	CheckFileFunc func(string) (fs.FileInfo, error)
 )
 
 var childProcPathFmt = "/proc/%[1]v/task/%[1]v/children"
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . nginxPlusClient
+
+type nginxPlusClient interface {
+	UpdateHTTPServers(
+		upstream string,
+		servers []ngxclient.UpstreamServer,
+	) (
+		added []ngxclient.UpstreamServer,
+		deleted []ngxclient.UpstreamServer,
+		updated []ngxclient.UpstreamServer,
+		err error,
+	)
+	GetUpstreams() (*ngxclient.Upstreams, error)
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . ProcessHandler
+
+type ProcessHandler interface {
```

Comment:
of course, thanks for patience with me

Reply:
I totally removed EnsureNginxRunning and idea is to use FindMainProcess(ctx context.Context,
	checkFile CheckFileFunc,
	readFile ReadFileFunc,
	timeout time.Duration) but keep in mind I need to pass all other args now

Reply:
@zedGGs having little issue currently wiith `checkFile CheckFileFunc` since it's being used here: https://github.com/nginxinc/nginx-gateway-fabric/pull/2175/commits/0ecf5d14e6ce45b0e1ae7770218b2dfb5789fa8e#diff-555c45daf604f58134a5937da621ca2d0c8900c924dfd21fabd417957ace5c82R214

Reply:
looks good now